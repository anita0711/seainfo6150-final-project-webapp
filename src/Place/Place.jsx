import React, { useEffect, useState } from "react";
import { isEmpty } from "lodash";
import PlaceDetails from "../PlaceDetails/PlaceDetails";
import Loading from "../images/loading.gif";
import styles from "./Place.module.css";

const Place = (props) => {
  const [fetchedData, setFetchedData] = useState({});
  useEffect(() => {
    const fetchData = async () => {
      const result = await fetch("https://demo3996095.mockable.io/places");
      const resultJson = await result.json();
      setFetchedData(resultJson);
    };

    if (isEmpty(fetchedData)) {
      fetchData();
    }
  }, [fetchedData]);

  let showPlaceDetails;
  if (!isEmpty(fetchedData)) {
    const place = fetchedData[props.category][props.placeId];
    showPlaceDetails = (
      <div>
        <div>
          <h2>{place.title}</h2>
        </div>
        <div>
          <PlaceDetails
            img1={place.img1._url}
            img2={place.img2._url}
            img3={place.img3._url}
            img4={place.img4._url}
            state={place.state}
            about={place.shortDescription}
            description={place.description}
            bestFor={place.bestFor}
            thingsToDo={place.thingsToDo}
            timeToVisit={place.timeToVisit}
            dining={place.dining}
            hotels={place.hotels}
            map={place.map}
          />
        </div>
      </div>
    );
  } else {
    showPlaceDetails = (
      <div>
        <img className={styles.loading} src={Loading} alt="loading" />
      </div>
    );
  }
  return showPlaceDetails;
};

export default Place;
