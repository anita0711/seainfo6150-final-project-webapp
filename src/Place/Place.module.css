.loading {
  display: block;
  margin-left: auto;
  margin-right: auto;
  width: 30%;
}

.wallpaper {
  background-image: url("../images/Background/Place-Details.jpg");
  background-repeat: no-repeat;
  background-size: cover;
}

.title {
  display: inline-block;
  background-color: rgb(209, 133, 83);
  color: white;
  border-radius: 10px;
  margin-bottom: 15px;
  margin-left: 10px;
  margin-top: 10px;
}
