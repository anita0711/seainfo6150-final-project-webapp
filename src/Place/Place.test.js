import React from "react";
import { render } from "@testing-library/react";
import Place from "./Place.jsx";

it("renders the Place component correctly", () => {
  const { container } = render(
    <Place category="summer" placeId="001-yellowstone" />
  );
  expect(container).toMatchSnapshot();
});
