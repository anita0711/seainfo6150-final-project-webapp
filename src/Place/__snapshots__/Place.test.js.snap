// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Place component correctly 1`] = `
<div>
  <div>
    <img
      alt="loading"
      class="loading"
      src="loading.gif"
    />
  </div>
</div>
`;
